database/tests/desktop/core/sync_point_spec_test.cc

```diff
@@ -0,0 +1,837 @@
+        EventRegistration* event_registration = nullptr;
+        int callback_id = spec->callbackId();
+        EXPECT_TRUE(callback_id != 0 && registrations.count(callback_id) != 0);
+        event_registration = registrations[callback_id];
+        std::vector<Event> actual = sync_tree_->RemoveEventRegistration(
+            event_registration->query_spec(),
+            static_cast<void*>(event_registration), kErrorNone);
+        EXPECT_THAT(actual, Pointwise(EventEq(), expected));
+        break;
+      }
+      case test_data::StepType_serverUpdate: {
+        Variant update = spec->data()
+                             ? FlexbufferToVariant(spec->data_flexbuffer_root())
+                             : Variant::Null();
+        std::vector<Event> actual;
+        if (spec->tag()) {
+          actual = sync_tree_->ApplyTaggedQueryOverwrite(path, update,
+                                                         Tag(spec->tag()));
+        } else {
+          actual = sync_tree_->ApplyServerOverwrite(path, update);
+        }
+        EXPECT_THAT(actual, UnorderedPointwise(EventEq(), expected));
+        break;
+      }
+      case test_data::StepType_serverMerge: {
+        std::map<Path, Variant> merges =
+            VariantToPathMap(FlexbufferToVariant(spec->data_flexbuffer_root()));
+        std::vector<Event> actual;
+        if (spec->tag()) {
+          actual =
+              sync_tree_->ApplyTaggedQueryMerge(path, merges, Tag(spec->tag()));
+        } else {
+          actual = sync_tree_->ApplyServerMerge(path, merges);
+        }
+        EXPECT_THAT(actual, UnorderedPointwise(EventEq(), expected));
+        break;
+      }
+      case test_data::StepType_set: {
+        Variant to_set = FlexbufferToVariant(spec->data_flexbuffer_root());
+        OverwriteVisibility visible =
+            spec->visible() ? kOverwriteVisible : kOverwriteInvisible;
+        // For now, assume anything visible should be persisted.
+        Persist persist = spec->visible() ? kPersist : kDoNotPersist;
+        std::vector<Event> actual = sync_tree_->ApplyUserOverwrite(
+            path, to_set, to_set, current_write_id++, visible, persist);
+        EXPECT_THAT(actual, UnorderedPointwise(EventEq(), expected));
+        break;
+      }
+      case test_data::StepType_update: {
+        CompoundWrite merges = CompoundWrite::FromVariantMerge(
+            FlexbufferToVariant(spec->data_flexbuffer_root()));
+        std::vector<Event> actual = sync_tree_->ApplyUserMerge(
+            path, merges, merges, current_write_id++, kPersist);
+        EXPECT_THAT(actual, UnorderedPointwise(EventEq(), expected));
+        break;
+      }
+      case test_data::StepType_ackUserWrite: {
+        int to_clear = static_cast<int>(spec->writeId());
+        AckStatus ack_status = spec->revert() ? kAckRevert : kAckConfirm;
+        std::vector<Event> actual =
+            sync_tree_->AckUserWrite(to_clear, ack_status, kPersist, 0);
```

Comment:
could you add a `/*{parameter_name}=*/0` for the last parameter here?

Reply:
Done